src/core/lib/event_engine/extensions/supports_fd.h

```diff
@@ -119,6 +119,23 @@ class EventEngineSupportsFdExtension {
   virtual std::unique_ptr<EventEngine::Endpoint> CreateEndpointFromFd(
       int fd, const EndpointConfig& config) = 0;
 
+  /// Creates an EventEngine::Endpoint from a file descriptor that is configured
+  /// and bound locally but not yet connected to a remote peer. Returns a
+  /// connection handle to track status of connection attempt. Created endpoint
```

Comment:
There is no API to track connection status, the handle is used to cancel connection attempts.